src/cmdline.rs

```diff
@@ -71,13 +72,48 @@ fn usage() -> Command {
 /// Parse the commandline into a `Command` to execute.
 pub fn parse() -> Command {
     trace!("parse");
-    let matches = get_app().get_matches();
-
+    let cwd = if let Ok(cwd) = env::current_dir() {
+        cwd
+    } else {
+        println!("sccache: Couldn't determine current working directory");
+        return usage();
+    };
     // The internal start server command is passed in the environment.
     let internal_start_server = match env::var("SCCACHE_START_SERVER") {
         Ok(val) => val == "1",
         Err(_) => false,
     };
+    let mut args: Vec<_> = env::args_os().collect();
+    if ! internal_start_server {
```

Comment:
You could just change things so that the SCCACHE_START_SERVER block above does an early-return. It'd mean it would override any other commands instead of printing the error message from below, but I think that's OK, since it's basically just an implementation detail.